hack/crd-schema-checker.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -euxo pipefail
+
+CHECKER=$INSTALL_DIR/crd-schema-checker
+
+TMP_DIR=$(mktemp -d)
+
+function cleanup {
+    rm -rf "$TMP_DIR"
+}
+
+trap cleanup EXIT
+
+
+for crd in config/crd/bases/*.yaml; do
+    mkdir -p "$(dirname "$TMP_DIR/$crd")"
+    git show "origin/$BRANCH:$crd" > "$TMP_DIR/$crd"
```

Comment:
Using origin/main instead of just main as a baseline is needed in the CI as there the local main already has the commit under test merged. However this can cause artifacts locally when somebody works on a non rebased commit. I.e. if origin/main has a new field, but the local commit is not rebased to origin/main then that field will appear as removed even though in reality the commit under test does not remove that field.

Reply:
It turned out that in prow CI we don't have refs like origin:main from the central repo. Instead prow exports env variables to know what is the base of the PR.